compiler/hash-reporting/src/report.rs

```diff
@@ -149,16 +150,95 @@ impl Report {
     pub fn is_warning(&self) -> bool {
         self.kind == ReportKind::Warning
     }
+
+    /// Add a general message to the [Report].
+    pub fn message(&mut self, message: impl ToString) -> &mut Self {
+        self.message = message.to_string();
+        self
+    }
```

Comment:
maybe name this `title()` since it is exclusive to all other messages that a report can contain